tests/test_restriction.cpp

```diff
@@ -0,0 +1,256 @@
+#include <gtest/gtest.h>
+#include "gmgpolar.h"
+class test_restriction : public ::testing::TestWithParam<int>
+{
+protected:
+    void SetUp() override
+    {
+        //initialize default parameters.
+        gyro::init_params();
+        gyro::icntl[Param::verbose]        = 0;
+        gyro::icntl[Param::debug]          = 0;
+        gyro::icntl[Param::extrapolation]  = 0;
```

Comment:
Why do you set this parameter if you don't use it anyway? If this, or others, are not used by the tested functions, do not set it to have it simple.